docs/reference/commands/run.rst

```diff
@@ -47,3 +47,18 @@ Update the application's source code before running. Equivalent to running::
 
     $ briefcase update
     $ briefcase run
+
+``--test``
+----------
+
+Run the test suite in the bundled app environment. Running ``run --test``
+forces an update and rebuild, to ensure that the packaged application
+contains all the test dependencies and code. To prevent this update and
+build, use ``-no-auto-update``.
+
+``--no-auto-update``
+--------------------
+
+Prevent the automated update that is performed when specifying by the
+``--test`` option. This option should only be required if you need to
+build your app on one machine, and run it on another.
```

Comment:
Sorry, I don't remember the details of why you needed this, can you explain again?

Reply:
It's an artefact of my Linux testing setup.

I've got a x86_64 macOS box, running Ubuntu 22.04 in a Parallels VM. I have to use 22.04 to check forward compatibility of binaries. Since it's in a VM, I can't run Docker locally. Since it's Ubuntu 22.04, I can't build AppImages with `--no-docker`. 

So - I need to build AppImages on the macOS side of the fence, inside the macOS Docker. However, I can't *run* the tests there. 

That means I need to be able to `build --test` on macOS, and then `run --test` on Linux. But `run --test` implies an update and build (to ensure we always have up to date test code), so we need a way to disable the update cycle when running --test.

When we eventually get source modification working and `--update` becomes redundant, I guess this option will become redundant as well; we'll be able to detect that nothing has changed since we did the build, and not do the forced update when in test mode. But for now, we need a way to disable the forced update implied by test mode.

Reply:
Even after we implement change detection (#807), I imagine there will be some situations where you might want to run the existing app without updating it. For example, you might be in the middle of fixing a bug, so your current source tree isn't runnable, but you want to run the previous build again to gather more information.

So how about renaming this option to `--no-update`, but otherwise keeping its implementation the same for now? Then once we have change detection, `--update` and `--no-update` can target the same `dest` variable, with `--update` being the default in both normal and test mode.

Reply:
That makes sense; and I've been able to consolidate the handling of `update` into a single argparse variable.